.github/workflows/release-check.yml

```
@@ -0,0 +1,21 @@
+name: Release Checker
+
+on:
+  pull_request_target:
+    paths: ["Cargo.toml"]
+    types: [ opened, synchronize, reopened, labeled, unlabeled ]
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  release-check:
+    uses: ipdxco/unified-github-workflows/.github/workflows/release-check.yml@v1.0
+    with:
+      sources: '["Cargo.toml"]'
```

That way, we would run the cargo publish dry run, only if the release-check created a draft GitHub Release. Please note, that the output that I reference doesn't exist yet, but I think it's a great idea to expose it.

```suggestion
  release-check:
    uses: ipdxco/unified-github-workflows/.github/workflows/release-check.yml@v1.0
    with:
      sources: '["Cargo.toml"]'
  cargo-publish-dry-run:
    needs: [release-check]
    if: steps.release-check.outputs.release != ''
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - run: cargo publish --dry-run
         shell: bash
```

---

👌 nice, this should be good; @Stebalien is this enough?

---

BTW, similarly, we could attach extra tag creation and cargo publishing to the releaser.yml.

---

This doesn't look like it handles multiple packages (we have multiple packages in the workspace). Honestly, I'm not actually sure how to dry-run with multiple packages.

See this thread: https://github.com/rust-lang/cargo/issues/1169. I was hoping that you might have found some magical solution...

(see https://github.com/rust-lang/cargo/issues/1169#issuecomment-1753461359)

---

OK, I think I'll get it now. I'm going to resolve this conversation and explain the changes I introduced in a comment below. Let me know what you think about them.